test_pytest/conftest.py

```diff
@@ -333,109 +333,128 @@ def nd_box(dim):
-    """returns True if Splines are equivalent"""
-    if a.whatami != b.whatami:
-        return False
-    for req_prop in a.required_properties:
-        if req_prop == "knot_vectors":
-            for aa, bb in zip(a.knot_vectors, b.knot_vectors):
-                if not np.allclose(aa.numpy(), bb.numpy()):
-                    if print_:
-                        print("a.kvs", a.kvs)
-                        print("b.kvs", b.kvs)
-                    return False
-        elif not np.allclose(getattr(a, req_prop), getattr(b, req_prop)):
-            if print_:
-                print(f"a.{req_prop}", getattr(a, req_prop))
-                print(f"b.{req_prop}", getattr(b, req_prop))
-            return False
-    return True
-
+def to_tmpf():
+    def _to_tmpf(tmpd):
+        """given tmpd, returns tmpf"""
+        return os.path.join(tmpd, "nqv248p90")
 
-@pytest.fixture
-def are_items_close(a, b):
-    """returns True if items in a and b are close"""
-    all_close = True
+    return _to_tmpf
 
-    for i, (aa, bb) in enumerate(zip(a, b)):
-        this_is_close = all(np.isclose(aa, bb))
-        if not this_is_close:
-            # print to inform
-            print(f"elements in index-{i} are not close")
-            print(f"  from first: {aa}")
-            print(f"  from second: {bb}")
 
-            all_close = False
+@pytest.fixture
+def are_splines_equal():
+    def _are_splines_equal(a, b, print_=False):
+        """returns True if Splines are equivalent"""
+        if a.whatami != b.whatami:
+            return False
+        for req_prop in a.required_properties:
+            if req_prop == "knot_vectors":
+                for aa, bb in zip(a.knot_vectors, b.knot_vectors):
+                    if not np.allclose(aa.numpy(), bb.numpy()):
+                        if print_:
+                            print("a.kvs", a.kvs)
+                            print("b.kvs", b.kvs)
+                        return False
+            elif not np.allclose(getattr(a, req_prop), getattr(b, req_prop)):
+                if print_:
+                    print(f"a.{req_prop}", getattr(a, req_prop))
+                    print(f"b.{req_prop}", getattr(b, req_prop))
+                return False
+        return True
 
-    return all_close
+    return _are_splines_equal
 
 
 @pytest.fixture
-def are_items_same(a, b):
-    """returns True if items in a and b are same"""
-    all_same = True
+def are_items_close():
+    def _are_items_close(a, b):
+        """returns True if items in a and b are close"""
+        all_close = True
 
-    for i, (aa, bb) in enumerate(zip(a, b)):
-        this_is_same = aa == bb
-        if not this_is_same:
-            # print to inform
-            print(f"element in index-{i} are not same")
-            print(f"  from first: {aa}")
-            print(f"  from second: {bb}")
+        for i, (aa, bb) in enumerate(zip(a, b)):
+            this_is_close = all(np.isclose(aa, bb))
+            if not this_is_close:
```

Comment:
```suggestion
            if not all(np.isclose(aa, bb)):
```

Why not do it like this?

Reply:
Done